### PR TITLE
Simplify abstract embedded

### PR DESCRIPTION
**Pull request by kellertuer**
I had a minute to spare and this already works with the manifolds actually.

### CODE REVIEW

**Comment by mateuszbaran** [comment]
I thought we were considering removing `embed!` for `AbstractEmbeddedManifold` entirely?

**Comment by kellertuer** [comment]
What do you further want to move? I thought those two lines it all it does?

**Comment by mateuszbaran** [comment]
Maybe let's discuss it on Slack :slightly_smiling_face: .

**Comment by mateuszbaran** [comment]
Looks nice, I just think now the docstring for `embed` should be corrected to say that:
  1. It is expected to be explicitly defined for `EmbeddedManifold`.
  2. It is defined as identity for `AbstractEmbeddedManifold`s.
  3. It may or may not be defined for other manifolds.

File: src/EmbeddedManifold.jl [hunk 1]
```diff
@@ -66,12 +70,14 @@ struct TransparentIsometricEmbedding <: AbstractIsometricEmbeddingType end
 """
     EmbeddedManifold{𝔽, MT <: Manifold, NT <: Manifold} <: AbstractDecoratorManifold{𝔽}
 
-A type to represent that a [`Manifold`](@ref) `M` of type `MT` is indeed an emebedded
-manifold and embedded into the manifold `N` of type `NT`.
-Points and tangent vectors are still represented in the representation of `M` and the
-manifold `M` is being decorated (contrary to [`AbstractEmbeddedManifold`](@ref)).
-To go to the representation in the embedding, the function [`embed`](@ref) should be used.
-[`project`](@ref) can be used to go the other way.
+A type to represent that a [`Manifold`](@ref) `M` of type `MT` can be embedded into a
```
Comment (mateuszbaran) [inline src/EmbeddedManifold.jl:73]: Maybe you could add that this type represents an explicit embedding of `M` into `N`? The phrase "can be embedded" is a bit vague.

File: src/EmbeddedManifold.jl [hunk 2]
```diff
@@ -66,12 +70,14 @@ struct TransparentIsometricEmbedding <: AbstractIsometricEmbeddingType end
...
+manifold `N` of type `NT`.
+
+!!!note
+    for a first (default) embedding, this type is not required, one can just implement
```
Comment (mateuszbaran) [inline src/EmbeddedManifold.jl:77]:
```suggestion
    This type is not required if a manifold `M` is to be embedded in one specific manifold `N`. One can then just implement
```

File: src/EmbeddedManifold.jl [hunk 3]
```diff
@@ -66,12 +70,14 @@ struct TransparentIsometricEmbedding <: AbstractIsometricEmbeddingType end
...
+    [`embed!`](@ref) and [`project!`](@ref), only for a second –maybe considered non-default–
```
Comment (mateuszbaran) [inline src/EmbeddedManifold.jl:78]:
```suggestion
    [`embed!`](@ref) and [`project!`](@ref). Only for a second –maybe considered non-default–
```

File: src/EmbeddedManifold.jl [hunk 4]
```diff
@@ -66,12 +70,14 @@ struct TransparentIsometricEmbedding <: AbstractIsometricEmbeddingType end
...
+    embedding, this type should be considered in order to dispatch on different embed
+    and project methods
```
Comment (mateuszbaran) [inline src/EmbeddedManifold.jl:80]:
```suggestion
    and project methods for different embeddings `N`.
```

File: src/ManifoldsBase.jl [hunk 5]
```diff
@@ -312,13 +312,17 @@ end
 """
     embed!(M::Manifold, q, p)
 
-Embed point `p` from the [`Manifold`](@ref) `M` into the ambient space and return the result in `q`.
+Embed point `p` from the [`Manifold`](@ref) `M` intointo an ambient space.
```
Comment (mateuszbaran) [inline src/ManifoldsBase.jl:315]:
```suggestion
Embed point `p` from the [`Manifold`](@ref) `M` into an ambient space.
```

**Review by mateuszbaran — APPROVED** [review]

**Pull request merged**